### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by manzitresor**
### Implemented Feature

- [ ] Add Redux Toolkit
- [ ] Create a directory that will contain all your Redux logic
- [ ] Configure a Redux store
- [ ] Define a slice of state for books
- [ ] Define a slice of state for categories

### CODE REVIEW

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
Hi @manzitresor 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request  ❌ ❌ ❌ 

- [ ] In your README, I noticed that the section `Future Features` is missing, please include 1-3 potential features to be added. 🤗

- [ ] For a descriptive readme, let's ensure that all main headers from the [README template](https://github.com/microverseinc/readme-template) are present in our repo.

* The only optional section is FAQ. ☺

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
+});
+
+export default store;
```
Comment (AmaduKamara) [inline src/redux/store.js]: - [ ] Could you please use the book and category reducers in this file by importing them and using them in the reducer object as required?  You've already created them and using them here makes them available outside where you're using any of the actions 👍🏽

```jsx
import booksReducer from "./books/booksSlice"
import categoriesReducer from "./categories/categoriesSlice"

const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})

export default store
```

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @manzitresor,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Used Redux Toolkit package 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by manzitresor** [comment]
Hello @AmaduKamara for review feedback.
I'm working on it .

**Review by topeogunleye — APPROVED** [review]
### Hi @manzitresor, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Used the book and category reducers in this file by importing them and using them in the reducer object as required. ✔️
 
## [Optional] suggestions:
- For future pull requests, please notice that your branch name `reducer_and_actions` does not follow the convention of being named after a feature or action. For example, you could name your branch `added_reducer_and_actions`. This makes it easier to track the changes and the purpose of each branch. 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**